pkg/osinfo/layer_scanner.go

```diff
@@ -117,12 +116,26 @@ func (loss *LayerScanner) extractFileFromTar(tarPath, filePath, destPath string)
 	}
 	defer f.Close()
 
+	// Read the first bytes to determine if the file is compressed
+	var sample [2]byte
+	var compressed bool
+	if _, err := io.ReadFull(f, sample[:]); err != nil {
+		return errors.Wrap(err, "sampling bytes from tar header")
```

Comment:
`sampling initial bytes from file` (it's not tar we are looking for ...)

Reply:
Rephrased to file :+1: